PEP8 in boots

dipy/boots/__init__.py:2:1: W391 blank line at end of file
dipy/boots/resampling.py:3:1: E265 block comment should start with '# '
dipy/boots/resampling.py:5:11: E401 multiple imports on one line
dipy/boots/resampling.py:12:80: E501 line too long (113 > 79 characters)
dipy/boots/resampling.py:12:99: E231 missing whitespace after ','
dipy/boots/resampling.py:14:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:21:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:21:27: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:21:29: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:21:38: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:21:40: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:21:52: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:21:54: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:71:28: E231 missing whitespace after ','
dipy/boots/resampling.py:75:9: E265 block comment should start with '# '
dipy/boots/resampling.py:79:58: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:79:60: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:81:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:81:21: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:81:23: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:81:29: E203 whitespace before ','
dipy/boots/resampling.py:81:37: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:81:39: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:81:49: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:81:51: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:117:5: E265 block comment should start with '# '
dipy/boots/resampling.py:126:5: E265 block comment should start with '# '
dipy/boots/resampling.py:135:5: E265 block comment should start with '# '
dipy/boots/resampling.py:140:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:145:11: E271 multiple spaces after keyword
dipy/boots/resampling.py:147:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:163:75: E203 whitespace before ','
dipy/boots/resampling.py:164:63: E127 continuation line over-indented for visual indent
dipy/boots/resampling.py:165:75: E203 whitespace before ','
dipy/boots/resampling.py:166:63: E127 continuation line over-indented for visual indent
dipy/boots/resampling.py:167:43: E202 whitespace before ')'
dipy/boots/resampling.py:170:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:170:27: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:170:29: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:193:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:199:13: E201 whitespace after '('
dipy/boots/resampling.py:200:45: E202 whitespace before ')'
dipy/boots/resampling.py:202:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:209:12: E128 continuation line under-indented for visual indent
dipy/boots/resampling.py:212:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:212:29: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:212:31: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:212:41: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:212:43: E251 unexpected spaces around keyword / parameter equals
dipy/boots/resampling.py:266:28: E231 missing whitespace after ','
dipy/boots/resampling.py:266:43: E261 at least two spaces before inline comment
dipy/boots/resampling.py:266:44: E262 inline comment should start with '# '
dipy/boots/resampling.py:268:10: E711 comparison to None should be 'if cond is None:'
dipy/boots/resampling.py:275:9: E265 block comment should start with '# '
dipy/boots/resampling.py:276:40: E203 whitespace before ':'
dipy/boots/resampling.py:278:9: E265 block comment should start with '# '
dipy/boots/resampling.py:281:52: E261 at least two spaces before inline comment
dipy/boots/resampling.py:281:53: E262 inline comment should start with '# '
dipy/boots/resampling.py:285:59: E128 continuation line under-indented for visual indent
dipy/boots/resampling.py:287:1: E302 expected 2 blank lines, found 1
dipy/boots/resampling.py:290:1: E302 expected 2 blank lines, found 1

--- a/dipy/boots/__init__.py
+++ b/dipy/boots/__init__.py
@@ -1,2 +1,1 @@
 # Init for core.stat dipy objects
-

--- a/dipy/boots/resampling.py
+++ b/dipy/boots/resampling.py
@@ -1,15 +1,21 @@
 #!/usr/bin/python
 
-#import modules
+# import modules
 import time
-import sys, os, traceback, optparse
+import sys
+import os
+import traceback
+import optparse
 import numpy as np
 import scipy as sp
 from copy import copy, deepcopy
 
 import warnings
 
-warnings.warn("This module is most likely to change both as a name and in structure in the future",FutureWarning)
+warningMsg = "This module is most likely to change both as\
+                a name and in structure in the future"
+warnings.warn(warningMsg, FutureWarning)
+
 
 def bs_se(bs_pdf):
     """
@@ -18,7 +24,8 @@ def bs_se(bs_pdf):
     N = len(bs_pdf)
     return np.std(bs_pdf) * np.sqrt(N / (N - 1))
 
-def bootstrap(x, statistic = bs_se, B = 1000, alpha = 0.95):
+
+def bootstrap(x, statistic=bs_se, B=1000, alpha=0.95):
     """
     Bootstrap resampling [1]_ to accurately estimate the standard error and
     confidence interval of a desired statistic of a probability distribution
@@ -68,17 +75,18 @@ def bootstrap(x, statistic = bs_se, B = 1000, alpha = 0.95):
         look at the jackknife. Ann. Stat. 7, 1-26.
     """
     N = len(x)
-    pdf_mask = np.ones((N,),dtype='int16')
+    pdf_mask = np.ones((N,), dtype='int16')
     bs_pdf = np.empty((B,))
 
     for ii in range(0, B):
-        #resample with replacement
+        # resample with replacement
         rand_index = np.int16(np.round(np.random.random(N) * (N - 1)))
         bs_pdf[ii] = statistic(x[rand_index])
 
-    return bs_pdf, bs_se(bs_pdf), abc(x, statistic, alpha = alpha)
+    return bs_pdf, bs_se(bs_pdf), abc(x, statistic, alpha=alpha)
 
-def abc(x, statistic = bs_se , alpha = 0.05, eps = 1e-5):
+
+def abc(x, statistic=bs_se, alpha=0.05, eps=1e-5):
     """
     Calculates the bootstrap confidence interval by approximating the BCa.
 
@@ -114,7 +122,7 @@ def abc(x, statistic = bs_se , alpha = 0.05, eps = 1e-5):
     ..  [2] DiCiccio, T.J., Efron, B., 1996. Bootstrap Confidence Intervals.
         Statistical Science. 11, 3, 189-228.
     """
-    #define base variables -- n, p_0, sigma_hat, delta_hat
+    # define base variables -- n, p_0, sigma_hat, delta_hat
     n = len(x)
     p_0 = np.ones(x.shape) / n
     sigma_hat = np.zeros(x.shape)
@@ -123,7 +131,7 @@ def abc(x, statistic = bs_se , alpha = 0.05, eps = 1e-5):
         sigma_hat[i] = __tt_dot(i, x, p_0, statistic, eps)**2
         delta_hat[i] = __tt_dot(i, x, p_0, statistic, eps)
     sigma_hat = (sigma_hat / n**2)**0.5
-    #estimate the bias (z_0) and the acceleration (a_hat)
+    # estimate the bias (z_0) and the acceleration (a_hat)
     a_hat = np.zeros(x.shape)
     a_num = np.zeros(x.shape)
     a_dem = np.zeros(x.shape)
@@ -132,17 +140,19 @@ def abc(x, statistic = bs_se , alpha = 0.05, eps = 1e-5):
         a_dem[i] = __tt_dot(i, x, p_0, statistic, eps)**2
     a_hat = 1 / 6 * a_num / a_dem**1.5
     z_0 = __calc_z0(x, p_0, statistic, eps, a_hat, sigma_hat)
-    #define helper variables -- w and l
+    # define helper variables -- w and l
     w = z_0 + __calc_z_alpha(1 - alpha)
     l = w / (1 - a_hat * w)**2
     return __tt(x, p_0 + l * delta_hat / sigma_hat, statistic)
+
 
 def __calc_z_alpha(alpha):
     """
     Classic "quantile function" that calculates inverse of cdf of standard
     normal.
     """
-    return  2**0.5 * sp.special.erfinv(2 * alpha - 1)
+    return 2**0.5 * sp.special.erfinv(2 * alpha - 1)
+
 
 def __calc_z0(x, p_0, statistic, eps, a_hat, sigma_hat):
     """
@@ -160,14 +170,15 @@ def __calc_z0(x, p_0, statistic, eps, a_hat, sigma_hat):
         b_hat[i] = __tt_dot_dot(i, x, p_0, statistic, eps)
         tt_dot[i] = __tt_dot(i, x, p_0, statistic, eps)
     b_hat = b_hat / (2 * n**2)
-    c_q_hat = (__tt(x, (1 - eps) * p_0 + eps * tt_dot / (n**2 * sigma_hat)
-                                                              , statistic) +
-               __tt(x, (1 - eps) * p_0 - eps * tt_dot / (n**2 * sigma_hat)
-                                                              , statistic) -
-               2 * __tt(x, p_0, statistic) ) / eps**2
+    c_q_hat = (__tt(x, (1 - eps) * p_0 + eps * tt_dot /
+                    (n**2 * sigma_hat), statistic) +
+               __tt(x, (1 - eps) * p_0 - eps * tt_dot /
+                    (n**2 * sigma_hat), statistic) -
+               2 * __tt(x, p_0, statistic)) / eps**2
     return a_hat - (b_hat / sigma_hat - c_q_hat)
 
-def __tt(x, p_0, statistic = bs_se):
+
+def __tt(x, p_0, statistic=bs_se):
     """
     Function that calculates desired statistic from observable data and a
     given proportional weighting.
@@ -190,14 +201,16 @@ def __tt(x, p_0, statistic = bs_se):
     """
     return statistic(x / p_0)
 
+
 def __tt_dot(i, x, p_0, statistic, eps):
     """
     First numerical derivative of __tt
     """
     e = np.zeros(x.shape)
     e[i] = 1
-    return ( (__tt(x, (1 - eps) * p_0 + eps * e[i], statistic) -
-              __tt(x, p_0, statistic)) / eps )
+    return ((__tt(x, (1 - eps) * p_0 + eps * e[i], statistic) -
+            __tt(x, p_0, statistic)) / eps)
+
 
 def __tt_dot_dot(i, x, p_0, statistic, eps):
     """
@@ -206,10 +219,11 @@ def __tt_dot_dot(i, x, p_0, statistic, eps):
     e = np.zeros(x.shape)
     e[i] = 1
     return (__tt_dot(i, x, p_0, statistic, eps) / eps +
-           (__tt(x, (1 - eps) * p_0 - eps * e[i], statistic) -
+            (__tt(x, (1 - eps) * p_0 - eps * e[i], statistic) -
             __tt(x, p_0, statistic)) / eps**2)
 
-def jackknife(pdf, statistic = np.std, M = None):
+
+def jackknife(pdf, statistic=np.std, M=None):
     """
     Jackknife resampling [3]_ to quickly estimate the bias and standard
     error of a desired statistic in a probability distribution function (pdf).
@@ -263,29 +277,31 @@ def jackknife(pdf, statistic = np.std, M = None):
            look at the jackknife. Ann. Stat. 7, 1-26.
     """
     N = len(pdf)
-    pdf_mask = np.ones((N,),dtype='int16') #keeps track of all n - 1 indexes
+    pdf_mask = np.ones((N,), dtype='int16')  # keeps track of all n - 1 indexes
     mask_index = np.copy(pdf_mask)
-    if M == None:
+    if M is None:
         M = N
     M = np.minimum(M, N - 1)
     jk_pdf = np.empty((M,))
 
     for ii in range(0, M):
         rand_index = np.round(np.random.random() * (N - 1))
-        #choose a unique random sample to remove
-        while pdf_mask[rand_index] == 0 :
+        # choose a unique random sample to remove
+        while pdf_mask[rand_index] == 0:
             rand_index = np.round(np.random.random() * (N - 1))
-        #set mask to zero for chosen random index so not to choose again
+        # set mask to zero for chosen random index so not to choose again
         pdf_mask[rand_index] = 0
         mask_index[rand_index] = 0
-        jk_pdf[ii] = statistic(pdf[mask_index > 0]) #compute n-1 statistic
+        jk_pdf[ii] = statistic(pdf[mask_index > 0])  # compute n-1 statistic
         mask_index[rand_index] = 1
 
-    return jk_pdf, (N - 1) * (np.mean(jk_pdf) - statistic(pdf)), np.sqrt(N -
-                                                          1) * np.std(jk_pdf)
+    return (jk_pdf, (N - 1) * (np.mean(jk_pdf) - statistic(pdf)),
+            np.sqrt(N - 1) * np.std(jk_pdf))
+
 
 def residual_bootstrap(data):
     pass
+
 
 def repetition_bootstrap(data):
     pass

--- a/dipy/boots/resampling.py
+++ b/dipy/boots/resampling.py
@@ -1,14 +1,8 @@
 #!/usr/bin/python
 
-# import modules
-import time
-import sys
-import os
-import traceback
-import optparse
 import numpy as np
 import scipy as sp
-from copy import copy, deepcopy
+
 
 import warnings
 
@@ -170,10 +164,10 @@ def __calc_z0(x, p_0, statistic, eps, a_hat, sigma_hat):
         b_hat[i] = __tt_dot_dot(i, x, p_0, statistic, eps)
         tt_dot[i] = __tt_dot(i, x, p_0, statistic, eps)
     b_hat = b_hat / (2 * n**2)
-    c_q_hat = (__tt(x, (1 - eps) * p_0 + eps * tt_dot /
-                    (n**2 * sigma_hat), statistic) +
-               __tt(x, (1 - eps) * p_0 - eps * tt_dot /
-                    (n**2 * sigma_hat), statistic) -
+    c_q_hat = (__tt(x, ((1 - eps) * p_0 + eps * tt_dot /
+                        (n**2 * sigma_hat)), statistic) +
+               __tt(x, ((1 - eps) * p_0 - eps * tt_dot /
+                        (n**2 * sigma_hat)), statistic) -
                2 * __tt(x, p_0, statistic)) / eps**2
     return a_hat - (b_hat / sigma_hat - c_q_hat)
 
@@ -208,8 +202,8 @@ def __tt_dot(i, x, p_0, statistic, eps):
     """
     e = np.zeros(x.shape)
     e[i] = 1
-    return ((__tt(x, (1 - eps) * p_0 + eps * e[i], statistic) -
-            __tt(x, p_0, statistic)) / eps)
+    return ((__tt(x, ((1 - eps) * p_0 + eps * e[i]), statistic) -
+             __tt(x, p_0, statistic)) / eps)
 
 
 def __tt_dot_dot(i, x, p_0, statistic, eps):
@@ -219,8 +213,8 @@ def __tt_dot_dot(i, x, p_0, statistic, eps):
     e = np.zeros(x.shape)
     e[i] = 1
     return (__tt_dot(i, x, p_0, statistic, eps) / eps +
-            (__tt(x, (1 - eps) * p_0 - eps * e[i], statistic) -
-            __tt(x, p_0, statistic)) / eps**2)
+            (__tt(x, ((1 - eps) * p_0 - eps * e[i]), statistic) -
+             __tt(x, p_0, statistic)) / eps**2)
 
 
 def jackknife(pdf, statistic=np.std, M=None):

